service/docs/source/runtime.rst

```diff
@@ -70,37 +76,73 @@ Runtime launch options.
+
+.. admonition:: Quick start for MPI applications
+
+   The ``geopmlaunch`` tool is the recommended user interface to launch the GEOPM Runtime for 
+   profiling and optimizing MPI applications. It
    wraps a launcher application (srun in this example), generates a summarizing
    report file, and optionally generates a time-series trace per host.
 
-.. admonition:: Quick Start
-
-  Run ``geopmlaunch`` with your application:
+   The steps for using ``geopmlaunch`` with your MPI application are:
 
   * Specify how many nodes and processes to use.
   * Run the ``geopmlaunch`` command wherever you would normally run the
-    wrapped command (e.g., ``srun``, ``mpiexec``, etc.).
+    wrapped launcher command (e.g., ``srun``, ``mpiexec``, etc.).
   * Read the generated ``geopm.report`` file
   
   .. code-block:: console
-    :caption: Examples using ``geopmlaunch``
+    :caption: examples using ``geopmlaunch``
 
-    $ # Launch with srun and explore the generated GEOPM report
-    $ geopmlaunch srun -N 1 -n 20 -- ./my-app
+    $ # launch with srun and explore the generated geopm report
+    $ geopmlaunch srun -n 1 -n 20 -- ./my-app
     $ less geopm.report
-    $ # Launch with Intel mpiexec and explore the generated GEOPM report
+    $ # launch with intel mpiexec and explore the generated geopm report
     $ geopmlaunch impi -n 1 -ppn 20 -- ./my-app
     $ less geopm.report
     $ # show all options and available launchers
     $ geopmlaunch --help
 
-  The `GEOPM Runtime tutorial
-  <https://github.com/geopm/geopm/tree/dev/tutorial#geopm-tutorial>`_ shows how
-  to profile unmodified applications, select and evaluate different GEOPM Agent
-  algorithms, and how to add markup to an application.  The tutorial provides a
-  starting point for someone trying to get familiar with the GEOPM Runtime.
 
+.. admonition:: Quick start for Non-MPI applications
+
+   In order to profile non-MPI application, the recommended approach is to launch
+   the GEOPM runtime alongside the target application. This can be done by launching
+   the ``geopmctl`` application in the background on every host node that the non-MPI 
+   application is expected to run on. After a clean or forced termination of the 
+   application being profiled, the ``geopmctl`` application is designed to terminate
+   and generate a summarizing report file, and optionally, a time-series trace per host. 
+ 
+   The following requirements must be met while launching ``geopmctl`` with your 
+   non-MPI application:
+ 
+   * Both the ``geopmctl`` process and the application process must have
+     the ``GEOPM_PROFILE`` environment variable set to the **same**
+     value.
+   * The application process must have ``LD_PRELOAD=libgeopm.so.1`` set
+     in the environment, or the application binary must be linked
+     directly to ``libgeopm.so.1`` at compile time.
+   * The ``GEOPM_REPORT`` environment variable must be set in the
```

Comment:
I'm thinking it would also be helpful to mention the `GEOPM_PROGRAM_FILTER` environment variable (not as a requirement, but as an optional feature that could help the user ensure the right application is profiled).

Reply:
Added text & code snippets to runtime.rst, overview.rst, and geopm.7.rst